### PR TITLE
[DinoMod] weak points with effects mk 2

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Mods "weak points with effects and requirements"

#### Purpose of change

Make combat more interesting and realistic, reward skilled players and different play styles

#### Describe the solution

Adds different weak point groups for several different dino types, plus pterosaurs and mosasaurs. Dinos with special armor get special armor bonuses on some weak points. Some weak points create effects and some require them. Changed all of them to use standard template weakpoint bodies. 

#### Describe alternatives you've considered

Scrap this and make more zombie dino evolutions

#### Testing

No errors, works as expected
Screen Shot 2021-11-23 at 3 53 11 PM

#### Additional context

Based on a ton of recent PRs building out weak points, especially #51961 . Second attempt to try and get the build to pass, rebuild of #52967

### CODE REVIEW

**Comment by LyleSY** [comment]
Still getting the same test failure. @dseguin can you confirm I'm not messing something up?

**Comment by dseguin** [comment]
It looks like your branch is based on a really old master (from November 22). You'll need to either rebase from the current CleverRaven/master or merge that master into this branch.

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+    "id": "wps_sauropod_body",
+    "weakpoints": [
+      {
+        "name": "the head",
+        "armor_mult": { "physical": 0.75 },
+        "crit_mult": { "all": 1.1 },
+        "difficulty": { "melee": 5, "ranged": 5 },
+        "coverage_mult": { "ranged": 0.75 },
+        "coverage": 3,
+        "required_effects": "downed",
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:78]:
```suggestion
        "required_effects": [ "downed" ],
```

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+        ]
+      },
+      {
+        "name": "the belly",
+        "armor_mult": { "physical": 0.5 },
+        "crit_mult": { "all": 1.2 },
+        "difficulty": { "melee": 4, "ranged": 1 },
+        "coverage_mult": { "ranged": 0.75 },
+        "coverage": 3,
+        "required_effects": "downed"
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:103]:
```suggestion
        "required_effects": [ "downed" ]
```

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+      },
+      {
+        "name": "the eye",
+        "armor_mult": { "physical": 0 },
+        "coverage": 1,
+        "crit_mult": { "all": 1.25 },
+        "coverage_mult": { "ranged": 0.5 },
+        "difficulty": { "point": 10, "broad": 10 },
+        "required_effects": "downed",
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:112]:
```suggestion
        "required_effects": [ "downed" ],
```

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+        ]
+      },
+      {
+        "name": "the belly",
+        "armor_mult": { "physical": 0.5 },
+        "crit_mult": { "all": 1.2 },
+        "difficulty": { "melee": 4, "ranged": 1 },
+        "coverage_mult": { "ranged": 0.75 },
+        "coverage": 3,
+        "required_effects": "downed"
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:233]:
```suggestion
        "required_effects": [ "downed" ]
```

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:298]:
```suggestion
        "required_effects": [ "downed" ]
```

File: data/mods/DinoMod/dinosaur_weakpoints.json [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
```
Comment (dseguin) [inline data/mods/DinoMod/dinosaur_weakpoints.json:363]:
```suggestion
        "required_effects": [ "downed" ]
```

**Review by dseguin — COMMENTED** [review]
Also, just some minor corrections: `"required_effects"` only accepts arrays, so `"downed"` would need to be defined as `[ "downed" ]`.

**Comment by LyleSY** [comment]
Bah, thought I did that when I recreated the PR, looks like I just moved the problem around

**Pull request merged**